pkg/boot/initrd.go

```diff
@@ -7,22 +7,31 @@ package boot
 import (
 	"bytes"
 	"io"
+	"strings"
 
 	"github.com/u-root/u-root/pkg/uio"
 )
 
-// CatInitrds concatenates initrds from a list of io.Readers, pads them to a
-// 512 byte boundary, and returns a uio.LazyOpenerAt
-func CatInitrds(initrds ...io.Reader) *uio.LazyOpenerAt {
-	return uio.NewLazyOpenerAt("", func() (io.ReaderAt, error) {
+// CatInitrds concatenates initrds on first ReadAt call from a list of
+// io.ReaderAts, pads them to a 512 byte boundary.
+func CatInitrds(initrds ...io.ReaderAt) io.ReaderAt {
+	var names []string
+	for _, initrd := range initrds {
+		names = append(names, stringer(initrd))
+	}
+
+	return uio.NewLazyOpenerAt(strings.Join(names, ", "), func() (io.ReaderAt, error) {
```

Comment:
Now this makes sense, I was a bit puzzled about the string there and left it for later. :)